### DISCUSSION TITLE
Inconsistency with nearest neighbour query - Ovito vs Freud (triclinic box)

### DISCUSSION THREAD

**Discussion by Gareth-Bellinger** [glotzerlab/freud#1244] [Support] [answered]
Hi there,

I have been using freud for the task of analyzing the number of nearest neighbours for each particle (Carbon atoms) with output files from LAMMPS and VASP. 

The last part of my analysis relies on identifying the number of nearest neighbours, but now the simulation box has been deformed by VASP during energy minimization. I implemented a check (via freud) to see if any carbon atom is bonded to more than 4 atoms initially just as a sanity check to tell me if files might need re-running on VASP or LAMMPS. This is when I noticed an inconsistency between my results from Freud and the results from Ovito. Namely, Freud said one carbon atom had 5 bonds total and for the same particle, Ovito said it only had 4. This occurs more than once for the set of the datafiles that I am analyzing. 

This prompted me to investigate the issue more systematically and I confirmed that it only occurred when I analyzed triclinic simulation boxes. Additionally, when I use the same cutoff radius in Freud and Ovito, I obtain a different number of total bonds for the same system.

I have included both an example CONTCAR (output file from VASP) and a portion of my python program that checks for the total bonding for the same cutoff radius in OVITO and Freud. I have also included the input file for VASP before it was deformed to triclinic geometry to show that the number of bonds are the same for a cubic box.

[ovito vs freud bonding files.zip](https://github.com/glotzerlab/freud/files/15132589/ovito.vs.freud.bonding.files.zip)

I am not sure where my error is, only that it might be related to 
a) I did not convert the output file coordinates (in reduced units, '[direct](https://www.vasp.at/wiki/index.php/POSCAR)') to cartesian properly and this is causing Freud to analyze positions that are not real
b) I should be using a different query type (i.e. not AABB)

I appreciate any help, thank you in advance!

**Comment by tommy-waltmann** [comment, marked as answer]
Hi Gareth,

The box data in the provided script is not being read the same way in both functions. When I print the `simbox_ovito` variable in the `ovito_analysis` function, I get the following:
```
[[12.83660914  0.38911115 -1.75788069  0.        ]
 [ 0.46766025 11.32910487  1.0593323   0.        ]
 [-2.10589517  1.13820917 12.10909819  0.        ]]
```

However, when I print the `box_data` variable from the `freud_analysis` function, I get the following:
```
[[12.83660914  0.46766025 -2.10589517]
 [ 0.38911115 11.32910487  1.13820917]
 [-1.75788069  1.0593323  12.10909819]]
```

They appear to be the transpose of one another. When I transpose the `box_data` matrix, the number of neighbors is the same again.

**Reply by Gareth-Bellinger** [reply]
Thank you and apologies for making a discussion on such a trivial error.
